libobs/audio-monitoring/nix/pulseaudio-wrapper.h

```diff
@@ -0,0 +1,178 @@
+ * the callback should be retained until the calling thread executes
+ * pulseaudio_accept()
+ *
+ * If wait_for_accept is 0 the function returns and the data is freed.
+ */
+void pulseaudio_signal(int wait_for_accept);
+
+/**
+ * Signal the waiting callback to return
+ *
+ * This function is used in conjunction with pulseaudio_signal()
+ */
+void pulseaudio_accept();
+
+/**
+ * Request source information
+ *
+ * The function will block until the operation was executed and the mainloop
+ * called the provided callback function.
+ *
+ * @return negative on error
+ *
+ * @note The function will block until the server context is ready.
+ *
+ * @warning call without active locks
+ */
+int_fast32_t
+pulseaudio_get_source_info_list(pa_source_info_cb_t cb, void *userdata);
+
+/**
+ * Request source information from a specific source
+ *
+ * The function will block until the operation was executed and the mainloop
+ * called the provided callback function.
+ *
+ * @param cb pointer to the callback function
+ * @param name the source name to get information for
+ * @param userdata pointer to userdata the callback will be called with
+ *
+ * @return negative on error
+ *
+ * @note The function will block until the server context is ready.
+ *
+ * @warning call without active locks
+ */
+int_fast32_t
+pulseaudio_get_source_info(pa_source_info_cb_t cb, const char *name,
+                           void *userdata);
+
+/**
+ * Request server information
+ *
+ * The function will block until the operation was executed and the mainloop
+ * called the provided callback function.
+ *
+ * @return negative on error
+ *
+ * @note The function will block until the server context is ready.
+ *
+ * @warning call without active locks
+ */
+int_fast32_t pulseaudio_get_server_info(pa_server_info_cb_t cb, void *userdata);
+
+/**
+ * Create a new stream with the default properties
+ *
+ * @note The function will block until the server context is ready.
+ *
+ * @warning call without active locks
+ */
+pa_stream *pulseaudio_stream_new(const char *name, const pa_sample_spec *ss,
+                                 const pa_channel_map *map);
+
+/**
+ * Connect to a pulseaudio playback stream
+ *
+ * @param s pa_stream to connect to. NULL for default
+ * @param attr pa_buffer_attr
+ * @param name Device name. NULL for default device
+ * @param flags pa_stream_flags_t
+ * @return negative on error
+ */
+int_fast32_t pulseaudio_connect_playback(pa_stream *s, const char *name,
+                                         const pa_buffer_attr *attr,
+                                         pa_stream_flags_t flags);
+
+/**
+ * Sets a callback function for when data can be written to the stream
+ *
+ * @param p pa_stream to connect to. NULL for default
+ * @param cb pa_stream_request_cb_t
+ * @param userdata pointer to userdata the callback will be called with
+ */
+void
+pulseaudio_write_callback(pa_stream *p, pa_stream_request_cb_t cb,
+                          void *userdata);
```

Comment:
Style: Indent with two tabs for continuing a function declaration.

This file needs to end with a newline.